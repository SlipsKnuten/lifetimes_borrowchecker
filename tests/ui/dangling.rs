fn bad_ref<'a>() -> &'a str {
    let s = String::from("temp");
    &s
}

fn main() {
    let _ = bad_ref();
}
