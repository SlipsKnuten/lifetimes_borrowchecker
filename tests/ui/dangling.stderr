error[E0515]: cannot return reference to local variable `s`
 --> tests/ui/dangling.rs:3:5
  |
3 |     &s
  |     ^^ returns a reference to data owned by the current function
